test/basic/test_parameters.jl

```diff
@@ -298,4 +298,83 @@ end
 end
 
 
+@testset "parameters -- prediction with early stopping" begin
```

Comment:
Ideally, this test should show for e.g. that there is 100 iterations for full prediction and 30 for early predictions as   adding output from later trees is not improving the predictions.
I found it difficult to access such information through the test so instead I went for comparing the cases with and without early stop where with a very low margin for early prediction stopping, these predictions will be less accurate and not matching the full predictions.
For regression there is no prediction early stopping and those parameters cannot be provided so the test for that can be removed.

Reply:
Removed tests with regressor because they were not needed